src/Phinx/Util/Util.php

```diff
@@ -82,10 +82,13 @@ public static function getExistingMigrationClassNames($path)
      */
     public static function getVersionFromFileName($fileName)
     {
+        $version = null;
```

Comment:
This violates the return Type. The exception is cleaner.

Reply:
If you look into the code the result is used directly downstream. So it would break anyway. A meaningful error message is more helpful here anyway. 
This is not a bc break then.

Reply:
The return type is already violated because it currently returns null when no match occurs. `$match` gets set to an empty array, thus the returned `$match[0]` does not exist, is coerced to null, and triggers a notice level error. This should not block this PR as compatibility is maintained. It should be marked nullable, i.e.:
```php
public static function getVersionFromFileName($fileName): ?string 
```
but that breaks compatibility for PHP <7.1.

I still believe changing the behaviour of this method is beyond the scope of this change.

Reply:
It's a public method, you cannot guarantee no bc break because application code may expect `null` without wrapping in `try` + `catch` blocks.

Reply:
As far as I can see. both internal uses of the method operate on file lists where the names have already been validated, hence it not being a problem in real-world use. I understand that this makes the code better, but I see no practical benefit untless the migration filename pattern changes.

Reply:
The code expects string. Lets keep it simple.

Reply:
> changing the behaviour of this method is beyond the scope of this change

You are already changing the behavior of phinx as a whole, making the class name more relaxed. At the same time we want to keep the version strictness (it currently had implicitly downstream). The exception at this point is warranted and currently needed due to your refactorings and phpstan noticing now.
If any library would build on this low level class then they can easily adjust their code. but it is very likely they do not need to as this use case is not normal procedure and the invalid file name would have already blown up somewhere else before anyway.